services/src/main/java/org/keycloak/authentication/authenticators/browser/IdentityProviderAuthenticator.java

```diff
@@ -66,8 +70,11 @@ private void redirect(AuthenticationFlowContext context, String providerId) {
                 String accessCode = new ClientSessionCode<>(context.getSession(), context.getRealm(), context.getAuthenticationSession()).getOrGenerateCode();
                 String clientId = context.getAuthenticationSession().getClient().getClientId();
                 String tabId = context.getAuthenticationSession().getTabId();
-                Response response = Response.seeOther(
-                        Urls.identityProviderAuthnRequest(context.getUriInfo().getBaseUri(), providerId, context.getRealm().getName(), accessCode, clientId, tabId))
+                URI location = Urls.identityProviderAuthnRequest(context.getUriInfo().getBaseUri(), providerId, context.getRealm().getName(), accessCode, clientId, tabId);
```

Comment:
Cool, so it even works with identity brokering. I assume this is only if it the external IdP is Keycloak as well?

Reply:
No, doesn't work yet with broker.  Its on my todo list and there' s a jira for it.

Reply:
If it doesn't work yet why are there changes to this class then? I don't think it has to work with brokering as that's really a web side flow so I would just drop that from the priority list.

Reply:
I actually don't know if it works or not.  I don't have a test for it.  I think it might work, but I haven't tried it out yet.

Reply:
We should forward the display type anyways, IMO.  Irregardless if this is supported or not.